driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -297,14 +296,16 @@ public Boolean isIdempotent() {
     @Override
     public String toString() {
         try {
-            if (forceNoValues) {
-                return getQueryString();
-            }
-            StringBuilder queryString = buildQueryString(null, CodecRegistry.DEFAULT_INSTANCE);
-            return maybeAddSemicolon(queryString).toString();
+            // try first with all values inlined (will not work if some values require custom codecs)
```

Comment:
I might be wrong, but since if `forceNoValues = true` then `getQueryString()` calls `buildQueryString(null, codecRegistry)`, it boils down to the same, except that we do not take advantage of the cache anymore (but is it important here?).